lib/framework/action.dart

```diff
@@ -359,6 +359,49 @@ class FileUploadAction extends EnsembleAction {
   }
 }
 
+class WalletConnectAction extends EnsembleAction {
+  WalletConnectAction({
+    this.id,
+    required this.wcProjectId,
+    required this.appName,
+    this.appDescription,
+    this.appUrl,
+    this.appIconUrl,
+    this.onComplete,
+    this.onError,
+  });
+
+  String? id;
+  String wcProjectId;
+  String appName;
+  String? appDescription;
+  String? appUrl;
+  String? appIconUrl;
+  EnsembleAction? onComplete;
+  EnsembleAction? onError;
+
+
+  factory WalletConnectAction.fromYaml({YamlMap? payload}) {
+    if (payload == null ||
+        (payload['wcProjectId'] == null && payload['name'] == null)) {
+      throw LanguageError(
```

Comment:
This doesn't correctly reflect the required-ness of both wcProjectId and appName above.

Reply:
payload['name'] doesn't exists.

Reply:
appMetaData is required but not enforced here